microprofile/WFLY-19592_MicroProfile_Fault_Tolerance_4_1.adoc

```diff
@@ -0,0 +1,147 @@
+:author:            Radoslav Husar
+:email:             rhusar@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The goal is to integrate MicroProfile (MP) Fault Tolerance (FT) 4.1 specification from the MP Platform 7.0 to WildFly at default stability level.
+The specification is implemented by SmallRye (SR) FT project.
+To certify functionality, the MP FT TCK is run as part of the WildFly testsuite.
+
+=== User Stories
+
+User wants to deploy an application using MicroProfile Fault Tolerance 4.1 from MicroProfile Platform 7.0.
+Users electing to employ MicroProfile Telemetry metrics will have MicroProfile Fault Tolerance metrics collected as well.
+
+== Issue Metadata
+
+* https://issues.redhat.com/browse/WFLY-19592[WFLY-19592 MicroProfile Fault Tolerance 4.1]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19588[WFLY-19588 Implement MicroProfile Platform 7] – MP Platform parent issue
+* https://issues.redhat.com/browse/WFLY-19590[WFLY-19590 MicroProfile Telemetry 2.0] – dependency of this issue for full Telemetry metrics integration
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly[WildFly]
+
+=== Other Interested Projects
+
+* https://github.com/eclipse/microprofile-fault-tolerance[Eclipse MicroProfile Fault Tolerance] – MP Specification and TCK sources
+* https://github.com/smallrye/smallrye-fault-tolerance[SmallRye Fault Tolerance] – implementation
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* Upgrade MP FT specification modules to 4.1.
+* Upgrade SmallRye FT to latest relevant version (6.6.x).
+* Enable collecting metrics with MP Telemetry when the subsystem is available in the server configuration.
+** Simultaneous metrics collection:
+Historically, when WildFly removed MP Metrics support,
+we have provided an optional Micrometer integration in SR FT in order to keep providing metrics collection support.
+This integration is not defined by the FT spec.
+The MP FT 4.1 spec defined MP Telemetry metrics collection.
+The spec also defines concurrent collection when both MP Metrics and MP Telemetry is available.
+Analogously, when both subsystems are available, we will collect metrics also with Micrometer.
+While this is helpful for instance as an upgrade path,
+simultaneous metrics collection might not be desirable in all situations.
+For that purpose, we will be supporting MP Config properties to fine-tune metrics collectors in use.
+These properties `smallrye.faulttolerance.micrometer.disabled` and `smallrye.faulttolerance.opentelemetry.disabled`
+are now supported by upstream SR FT (since SR FT 6.6.3 and 6.7.0) and documented.
+Note, that this is in addition to an existing spec-defined MP Config property to disable metrics collection by MP FT altogether.
```

Comment:
Thanks @rhusar, this is a nice additional feature.

Contents LGTM, formatting left to your own judgement (I see you used new lines in the raw contents, but it's rendered as a unique paragraph). As said, just formatting stuff, and it might be what you wanted.
I am wondering whether we should mention any related WFLY issue to have the tracker here, WDYT?

Reply:
> Contents LGTM, formatting left to your own judgement (I see you used new lines in the raw contents, but it's rendered as a unique paragraph)

Yes, that's intentional as a 'subtopic'.

> I am wondering whether we should mention any related WFLY issue to have the tracker here, WDYT?

All that information is better available from Jira where I have linked the issues (I like to keep my component upgrades separates as that greatly helps with tracking what happened and when). Thanks for the reminder.

Reply:
@rhusar This looks great to me.

Reply:
@rhusar Yes, sounds good. Thanks!

Reply:
Thanks everyone! Resolving this convo.